lib/components/src/ScrollBar/ScrollArea.tsx

```diff
@@ -0,0 +1,344 @@
+    enableHorizontal,
+    scrollRef,
+    horizontalPosition,
+    verticalPosition,
+    sliderPadding,
+    sliderSize,
+  ]);
+
+  const handleScroll = useCallback(
+    (event: React.UIEvent<HTMLDivElement, UIEvent>) => {
+      const { scrollLeft, scrollTop } = event.currentTarget;
+      const oldScrollRef = scrollRef.current.i;
+      scrollRef.current.i += 1;
+
+      const oldScrollLeft = scrollRef.current.left;
+      const oldScrollTop = scrollRef.current.top;
+      scrollRef.current.top = scrollTop;
+      scrollRef.current.left = scrollLeft;
+
+      const verticalScrollChange = scrollTop !== oldScrollTop;
+      const horizontalScrollChange = scrollLeft !== oldScrollLeft;
+
+      const allowVertical = innerRect.height > outerRect.height && enableVertical;
+      const allowHorizontal = innerRect.width > outerRect.width && enableHorizontal;
+
+      setState({
+        ...state,
+        horizontal: {
+          ...state.horizontal,
+          ...getHorizontalState({
+            enableHorizontal,
+            enableVertical: allowVertical,
+            horizontalPosition,
+            innerRect,
+            outerRect,
+            scrollLeft,
+            sliderPadding,
+            sliderSize,
+            verticalPosition,
+          }),
+          show: showOn === 'scroll' ? horizontalScrollChange : state.horizontal.show,
+        },
+        vertical: {
+          ...state.vertical,
+          ...getVerticalState({
+            enableHorizontal: allowHorizontal,
+            enableVertical,
+            horizontalPosition,
+            innerRect,
+            outerRect,
+            scrollTop,
+            sliderPadding,
+            sliderSize,
+            verticalPosition,
+          }),
+          show: showOn === 'scroll' ? verticalScrollChange : state.vertical.show,
+        },
+      });
+
+      if (onScroll) {
+        event.persist();
+        onScroll(event);
+      }
+
+      if (showOn === 'scroll') {
+        setTimeout(() => {
+          if (oldScrollRef + 1 === scrollRef.current.i) {
+            scrollRef.current.i = 0;
+            setState({
+              ...state,
+              horizontal: { ...state.horizontal, show: false },
+              vertical: { ...state.vertical, show: false },
+            });
+          }
+        }, sliderFadeout);
+      }
+    },
+    [
+      enableHorizontal,
+      enableVertical,
+      getHorizontalState,
+      getVerticalState,
+      horizontalPosition,
+      innerRect,
+      outerRect,
+      outerRef,
+      setState,
+      showOn,
+      sliderFadeout,
+      sliderPadding,
+      sliderSize,
+      state,
+      verticalPosition,
+    ]
+  );
+
+  const handleMouseEnter = useCallback(
+    (event) => {
+      if (showOn === 'hover') {
+        setState({
+          ...state,
+          horizontal: { ...state.horizontal, show: true },
+          vertical: { ...state.vertical, show: true },
+        });
+      }
+
+      if (rest.onMouseEnter) {
```

Comment:
shouldn't this be just handled as `onMouseEnter` prop?

Reply:
I am not interested in the onMouseEnter prop from the outside world myself

And since the  `<Styled.Wrapper />` is the one to get the spreaded props (which would include the onMouseEnter) but the `<Styled.ScrollableContainer />` is the one to actually hold the event I just give the option of having a "native" feel to the event handling in this case, without breaking anything.

Use still gets his onMouseEnter - but due to some fancy DOM setup it has to be derived from another actual DOM element

Reply:
Hmmmm - you got me thinking... will get back to you here 😄

Reply:
You are right - fixed